fix: entites out of boundaries material on remove

pbosio opened the pull request:
In some cases, using debug (like builder), if a mesh is removed while out of boundaries it may stay cached with the red "out of boundaries" material instead of it original one.

Commit:
fix: entites out of boundaries material on remove

Commit:
added nullcheck

BrianAmadori reviewed (changes requested):
Cool! We should add test to cover this use case and its good to go.

pbosio commented:
> Cool! We should add test to cover this use case and its good to go.

I've tried to create a test for that already, but I couldn't reproduce the issue with the test 🤷‍♂️

Commit:
add test + test fix

- had to move the RemoveEntityToBeChecked before entity.Cleanup, because the entity.gameObject is null when trying to reset the material

Commit:
test fixes

Commit:
Merge branch 'master' into fix/entity-removed-out-boundaries
